dao/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.dao.constant.sql.UserSql;
+import com.epam.esm.entity.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public List<User> findAll() {
+        Session session = sessionFactory.openSession();
+        return (List<User>) session.createSQLQuery(UserSql.FIND_ALL).addEntity(User.class).list();
```

Comment:
Please, try to use more high-level `session` methods (like `find`, ...) instead of using SQL. Your task is to practice hibernate, not SQL

Reply:
I don’t know a more global way besides HQL,JPQL,Criteria API. What should i use?